### DISCUSSION TITLE
Using OpenSSL 3.0.x with OpenSSL 3.0.9 FIPS module if x < 9

### DISCUSSION THREAD

**Discussion by kulkarniamit** [openssl/openssl#25048] [Q&A] [answered]
According to [README-FIPS.md](https://github.com/openssl/openssl/blob/master/README-FIPS.md#download-and-build-the-latest-release-of-openssl), 
"_Download and build the latest release of OpenSSL - We use OpenSSL 3.1.0 here, (but you could also use the latest 3.0.X)_"

So, latest releases of 3.x or 3.0.x can be used with 3.0.9 FIPS provider.

Can we also use 3.0.9 FIPS provider with a lower version of OpenSSL 3.0.x?
For example: Can we use OpenSSL 3.0.1 binary and static libaries with `fips.so` from OpenSSL 3.0.9?

_NOTE: I don't plan to use old versions of 3.0.x but trying to understand the compatibility._

**Comment by t-j-h** [comment, marked as answer]
The ABI is deliberately set so any 3.x fips module works with any 3.x.y in general.

**Reply by kulkarniamit** [reply]
Thank you for clarifying that 👍🏼
